app/web/components/UserSummary.tsx

```diff
@@ -77,25 +78,40 @@ export default function UserSummary({
     },
   );
 
+  const nameValue = user
+    ? nameOnly
+      ? user.name
+      : `${user.name}, ${user.age}`
+    : "";
+
+  const cityValue = user?.city ?? "";
+  const displayName =
+    nameValue.length > 20 ? nameValue.slice(0, 20) + "..." : nameValue;
+
+  const displayCity =
+    cityValue.length > 45 ? cityValue.slice(0, 45) + "..." : cityValue;
+
   const title = (
-    <Typography
-      component={headlineComponentWithRef}
-      variant="h2"
-      noWrap={nameOnly}
-      sx={{ marginTop: "auto", fontSize: "1.2rem" }}
-    >
-      {!user ? (
-        <Skeleton
-          data-testid={USER_TITLE_SKELETON_TEST_ID}
-          sx={{ maxWidth: 300 }}
-        />
-      ) : (
-        <>
-          {nameOnly ? user.name : `${user.name}, ${user.age}`}
-          {user.hasStrongVerification ? <StrongVerificationBadge /> : null}
-        </>
-      )}
-    </Typography>
+    <Tooltip title={nameValue} arrow placement="top">
+      <Typography
+        component={headlineComponentWithRef}
+        variant="h2"
+        noWrap={nameOnly}
+        sx={{ marginTop: "auto", fontSize: "1.2rem" }}
+      >
+        {!user ? (
+          <Skeleton
+            data-testid={USER_TITLE_SKELETON_TEST_ID}
+            sx={{ maxWidth: 300 }}
+          />
+        ) : (
+          <>
+            <span>{displayName}</span>
+            {user.hasStrongVerification ? <StrongVerificationBadge /> : null}
+          </>
+        )}
+      </Typography>
+    </Tooltip>
```

Comment:
Would you mind breaking this into a separate PR since it's for a different ticket?